### DISCUSSION TITLE
Run Photoview on RaspberryPi or Odroid (arm computers)

### DISCUSSION THREAD

**Discussion by E5tian** [photoview/photoview#177] [Q&A and Troubleshooting]
Thank you for your great project. This is exactly what I was looking for.
But unfortunately I have a problem that I cannot run photview on my Odroid (arm) or RaspberryPi. I followed the installation instructions in the ReadMe. But there is the first problem that the database used is not available for arm computers. So I adapted the docker-composer.yml file as follows. 

version: "3"

services:
  db:
    image: linuxserver/mariadb:arm32v6-latest
    restart: always
    environment:
      - MYSQL_DATABASE=photoview
      - MYSQL_USER=photoview
      - MYSQL_PASSWORD=photosecret
      - MYSQL_RANDOM_ROOT_PASSWORD=1
    volumes:
      - db_data:/var/lib/mysql

  photoview:
    image: viktorstrate/photoview:1
    restart: always
    ports:
      - "8000:80"
    depends_on:
      - db

    environment:
      - MYSQL_URL=photoview:photosecret@tcp(db)/photoview
      - API_LISTEN_IP=photoview
      - API_LISTEN_PORT=80
      - PHOTO_CACHE=/app/cache

      # Change This: The publicly exposed url
      # For example if the server is available from the domain example.com,
      # change this value to http://example.com/
      - PUBLIC_ENDPOINT=http://localhost:8000/

      # Optional: To enable map related features, you need to create a mapbox token.
      # A token can be generated for free here https://account.mapbox.com/access-tokens/
      # It's a good idea to limit the scope of the token to your own domain, to prevent others from using it.
      # - MAPBOX_TOKEN=<YOUR TOKEN HERE>

    volumes:
      - api_cache:/app/cache

      # Change this to the directory where your photos are located on your server.
      # If the photos are located at `/home/user/photos`, then change this value
      # to the following: `/home/user/photos:/photos:ro`.
      # You can mount multiple paths, if your photos are spread across multiple directories.
      - ./photos_path:/photos:ro

volumes:
  db_data:
  api_cache:

Then I can start the Docker container successfully. But unfortunately I cannot reach photview via http://localhost: 8000 afterwards.
What am I doing wrong?

I would be very happy to receive help or ideas.

**Comment by viktorstrate** [comment]
Hi @E5tian version 2 of Photoview has just been released, but it looks like there's some problems with the ARM build (see #179).
I would suggest that you wait a little until we fix the ARM builds for v2 instead of configuring v1, as you won't be able to upgrade later.

**Reply by E5tian** [reply]
Thank you for the quick response :-) I'll wait a bit .

**Reply by viktorstrate** [reply]
@E5tian version 2 has been released now!
Try again with the [newest docker-compose.example.yml](https://github.com/photoview/photoview/blob/master/docker-compose.example.yml)

**Reply by E5tian** [reply]
Thank you for your message. I tested it yesterday but unfortunately I cannot reach photoview either in the network or on the localhost. I think it's due to the 32bit OS of the Raspberry Pi 3. I'll try again on a Pi 4.
Thank you for your support :-)

**Reply by viktorstrate** [reply]
Okay, if you need help feel free to join the Discord server. There's a link in the readme.
